### PR TITLE
Swaps: logic, data flow, and performance fixes

### PR DESCRIPTION
**Pull request by christianbaroni**
## What changed (plus any additional context for devs)
- _Will add details in a bit_

## What to test

Fixes APP-1463
Fixes APP-1523
Fixes APP-1513
Fixes APP-1502
Fixes APP-1492
Fixes APP-1505
Fixes APP-1536
Fixes APP-1535
Fixes APP-1352
Fixes APP-1488
Fixes APP-1330

### CODE REVIEW

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 1]
```diff
@@ -31,33 +29,70 @@ import {
 } from '@/resources/assets/externalAssetsQuery';
 import { ethereumUtils } from '@/utils';
 import { queryClient } from '@/react-query';
+import { userAssetsStore } from '@/state/assets/userAssets';
+
+function omitNativePriceWorklet(
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:34]: for discussion: we should split out the animated reactions and remove this

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 2]
```diff
@@ -31,33 +29,70 @@ import {
...
+    niceIncrementFormatter({
+      incrementDecimalPlaces: initialDecimalPlaces,
+      inputAssetBalance: initialBalance,
+      inputAssetUsdPrice: initialSelectedInputAsset?.price?.value ?? 0,
+      niceIncrement: initialNiceIncrement,
+      percentageToSwap: 0.5,
+      sliderXPosition: SLIDER_WIDTH / 2,
+    })
+  );
+  const initialInputNativeValue = addCommasToNumber((initialInputAmount * (initialSelectedInputAsset?.price?.value ?? 0)).toFixed(2));
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:60]: we should have native formatter functions with the bigint work Bruno is doing

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 3]
```diff
@@ -31,33 +29,70 @@ import {
...
+  outputProgress: SharedValue<number>;
   quote: SharedValue<Quote | CrosschainQuote | QuoteError | null>;
+  sliderXPosition: SharedValue<number>;
 }) {
+  const { initialInputAmount, initialInputNativeValue } = getInitialInputValues();
+
   const inputValues = useSharedValue<{ [key in inputKeys]: number | string }>({
-    inputAmount: 0,
-    inputNativeValue: 0,
+    inputAmount: initialInputAmount,
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:94]: open question / note to self: what's happening for the scenario for internal deeplinks if we want to "BUY" a specific token

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 4]
```diff
@@ -73,8 +108,15 @@ export function useSwapInputsController({
   });
   const incrementDecimalPlaces = useDerivedValue(() => countDecimalPlaces(niceIncrement.value));
 
+  const inputNativePrice = useDerivedValue(() => {
+    return internalSelectedInputAsset.value?.nativePrice || internalSelectedInputAsset.value?.price?.value || 0;
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:112]: note to self: remind me the difference between these sources of prices?

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 5]
```diff
@@ -171,10 +220,14 @@ export function useSwapInputsController({
...
+      inputPrice,
+      outputPrice,
     }: {
       data: Quote | CrosschainQuote | QuoteError | null;
       outputAmount?: number;
       inputAmount?: number;
+      inputPrice?: number | null;
+      outputPrice?: number | null;
     }) => {
       'worklet';
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:233]: for discussion: we're doing this slider update in multiple places - what if we turned the slider into an animated reaction and react when: the input amount changes and the input method is not the slider

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 6]
```diff
@@ -193,110 +246,93 @@ export function useSwapInputsController({
         }
       }
 
-      isFetching.value = false;
       quote.value = data;
-      runOnJS(updateQuoteStore)(data);
 
       if (!data || (data as QuoteError)?.error) {
         return;
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:252]: - when there is no data or if there is an error, we are returning here without updating the isFetching.value
- as for runOnJS(updateQuoteStore): who needs the quote store? and if there was an error, do we want to update the store accordingly or no?

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 7]
```diff
@@ -193,110 +246,93 @@ export function useSwapInputsController({
...
       }
 
       if (inputAmount) {
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:255]: we now fetch asset prices along with the quote - we get price data for both input and output assets, but the quote will only return an inputAmount or outputAmount depending on what type of quote it was.

below, we are only updating the price for the input or output asset depending on the type of the quote (these are separate things, the prices for both assets can be updated)

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 8]
```diff
@@ -193,110 +246,93 @@ export function useSwapInputsController({
...
-        const price = internalSelectedInputAsset.value?.nativePrice || internalSelectedInputAsset.value?.price?.value || 0;
+        const price = inputPrice || inputNativePrice.value;
         inputValues.modify(prev => {
+          updateNativePriceForAsset({ price, type: 'inputAsset' });
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:258]: additionally: why is this `updateNativePriceForAsset` inside the modify function itself of the `inputValues`; everything here is being run synchronously on the UI thread so not sure why it needs to be inside the `modify` (same for outputAmount below)

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 9]
```diff
@@ -171,10 +220,14 @@ export function useSwapInputsController({
       data,
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:220]: I liked Christian's idea of appending `Worklet` to worklet functions - leaving a note here for `setQuote`

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 10]
```diff
@@ -193,110 +246,93 @@ export function useSwapInputsController({
...
 
       if (inputAmount) {
-        const price = internalSelectedInputAsset.value?.nativePrice || internalSelectedInputAsset.value?.price?.value || 0;
+        const price = inputPrice || inputNativePrice.value;
         inputValues.modify(prev => {
+          updateNativePriceForAsset({ price, type: 'inputAsset' });
           return {
             ...prev,
-            inputAmount,
+            inputAmount: Number(inputAmount),
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:261]: wondering why we're doing an extra cast to `Number` when `inputAmount` is already a number type

File: src/__swaps__/screens/Swap/components/SearchInput.tsx [hunk 11]
```diff
@@ -35,30 +52,57 @@ export const SearchInput = ({
...
+
+  const searchInputValue = useAnimatedProps(() => {
     // Removing the value when the input is focused allows the input to be reset to the correct value on blur
-    const query = isFocused ? undefined : defaultValue;
+    const query = isSearchFocused.value ? undefined : '';
 
     return {
-      defaultValue,
       text: query,
       selectionColor: getColorValueForThemeWorklet(asset.value?.highContrastColor, isDarkMode, true),
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/SearchInput.tsx:93]: unrelated to this PR, but this is causing a crash on Android.

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 12]
```diff
@@ -193,110 +246,93 @@ export function useSwapInputsController({
...
     },
     [
+      inputNativePrice,
       inputValues,
-      internalSelectedInputAsset.value?.balance.amount,
-      internalSelectedInputAsset.value?.nativePrice,
-      internalSelectedInputAsset.value?.price?.value,
-      internalSelectedOutputAsset.value?.nativePrice,
-      internalSelectedOutputAsset.value?.price?.value,
+      internalSelectedInputAsset,
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:287]: why not keep the specific data we need in this function which is now just the `internalSelectedInputAsset.value?.balance.amount` ?

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 13]
```diff
@@ -315,45 +351,45 @@ export function useSwapInputsController({
             type: 'outputAsset',
           },
         ].map(getAssetNativePrice)
-      );
+      ).then(([inputPrice, outputPrice]) => ({ inputPrice, outputPrice }));
     },
     [getAssetNativePrice]
   );
 
   const fetchAndUpdateQuote = async ({
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:359]: can rename this function as this now does more than just fetch the quote (e.g, also fetches prices)

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 14]
```diff
@@ -315,45 +351,45 @@ export function useSwapInputsController({
...
+    const resetFetchingStatus = ({ fromError = false }: { fromError?: boolean } = {}) => {
       'worklet';
-      isQuoteStale.value = 0;
       isFetching.value = false;
+      isQuoteStale.value = 0;
 
       // NOTE: start the quote fetching interval if the token lists aren't open
       // we need this check here because the user can open and close the token list before the quote is fetched and updated
       if (inputProgress.value <= NavigationSteps.INPUT_ELEMENT_FOCUSED && outputProgress.value <= NavigationSteps.INPUT_ELEMENT_FOCUSED) {
-        quoteFetchingInterval.start();
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:340]: note to self: to make sure I understand this

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 15]
```diff
@@ -367,69 +403,101 @@ export function useSwapInputsController({
...
-      data: response,
-    });
-
-    // TODO: Handle native asset inputs
-    runOnUI(setQuote)({
-      data: response,
-      outputAmount:
-        lastTypedInput === 'inputAmount'
+    try {
+      const [quoteResponse, fetchedPrices] = await Promise.all([
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:422]: what if one or the other fails (e.g, quote succeeds, but price fetching fails)

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 16]
```diff
@@ -367,69 +403,101 @@ export function useSwapInputsController({
...
+        params.swapType === SwapType.crossChain ? getCrosschainQuote(params) : getQuote(params),
+        fetchAssetPrices({
+          inputAsset: internalSelectedInputAsset.value,
+          outputAsset: internalSelectedOutputAsset.value,
+        }),
+      ]);
+
+      const quotedInputAmount =
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:430]: this parsing of the `quoteResponse` for `quotedInputAmount` and `quotedOutputAmount` could live in `setQuote` since it's receiving the `quoteResponse` as well as these params anyway

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 17]
```diff
@@ -367,69 +403,101 @@ export function useSwapInputsController({
...
+        lastTypedInputParam === 'outputAmount'
           ? Number(
               convertRawAmountToDecimalFormat(
-                (response as Quote)?.buyAmountMinusFees?.toString(),
-                internalSelectedOutputAsset.value?.decimals || 18
+                (quoteResponse as Quote)?.sellAmount?.toString(),
+                internalSelectedInputAsset.value?.decimals || 18
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:435]: I've always hated this decimal fallback to 18

File: src/navigation/Routes.ios.tsx [hunk 18]
```diff
@@ -299,7 +284,7 @@ function NativeStackNavigator() {
       <NativeStack.Screen component={AppIconUnlockSheet} name={Routes.APP_ICON_UNLOCK_SHEET} {...appIconUnlockSheetConfig} />
       <NativeStack.Screen component={ControlPanel} name={Routes.DAPP_BROWSER_CONTROL_PANEL} {...dappBrowserControlPanelConfig} />
 
-      {swapsV2Enabled && <NativeStack.Screen component={SwapsNavigator} name={Routes.SWAP_NAVIGATOR} {...swapConfig} />}
+      {swapsV2Enabled && <NativeStack.Screen component={SwapScreen} name={Routes.SWAP_NAVIGATOR} {...swapConfig} />}
```
Comment (walmat) [inline src/navigation/Routes.ios.tsx:287]:
```suggestion
      {swapsV2Enabled && <NativeStack.Screen component={SwapScreen} name={Routes.SWAP} {...swapConfig} />}
```

If we're going to only have a single screen, let's just route to the SWAP route

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 19]
```diff
@@ -367,69 +403,101 @@ export function useSwapInputsController({
...
             )
-          : undefined,
-    });
-    runOnUI(resetFetchingStatus)((response as QuoteError)?.error);
+          : undefined;
+
+      // TODO: Handle native asset inputs
+      runOnUI(() => {
+        // Check whether the quote has been superseded by new user input so we don't introduce conflicting updates
+        const isLastTypedInputStillValid = lastTypedInputParam === lastTypedInput.value;
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:453]: this logic below for checking of the quote is still valid should be the responsibility of the `setQuote` worklet which should be passed the original quote params to compare the current state against.

`setQuote` could be renamed to something like `validateAndSetQuoteWorklet` or `parseAndSetQuoteWorklet` etc
    
below, we are taking these from the params (passed into `buildQuoteParams`)
- lastTypedInputParam
- inputAmount
- outputAmount

and comparing it to the live state of:
- lastTypedInput.value
- inputValues.value.inputAmount
- inputValues.value.outputAmount

If we want to be thorough, we should compare it to the full params of the quote request.
Basically, we want to base it off of the params to `buildQuoteParams`
Edge case scenario: `lastTypedInputParam` is `inputAmount` and the value is `5`; we switch assets but the input amount is still the same

File: src/navigation/onNavigationStateChange.js [hunk 20]
```diff
@@ -50,6 +50,7 @@ export function onHandleStatusBar(currentState, prevState) {
     case Routes.DAPP_BROWSER_SCREEN:
     case Routes.WELCOME_SCREEN:
     case Routes.CHANGE_WALLET_SHEET:
+    case Routes.SWAP_NAVIGATOR:
```
Comment (walmat) [inline src/navigation/onNavigationStateChange.js:53]:
```suggestion
```

File: src/__swaps__/screens/Swap/components/UserAssetsSync.tsx
Comment (walmat) [inline src/__swaps__/screens/Swap/components/UserAssetsSync.tsx:1]: I'm not sure I understand a few of these checks/functionality in this file.

When the user switches wallets, shouldn't `currentAddress` reflect the new wallet. So basically we're only fetching user assets **once** here? Is that intended? I feel like this is problematic because what happens if a user gets sent an asset while staying on the same wallet. Currently, that asset would never be brought in.

I think we want this hook to always be enabled when `!!currentAddress`. But we only want to update certain things under certain circumstances inside of the `onSuccess` callback.

Sorry if I'm wrong here, maybe I don't have certain context that is needed.

File: src/__swaps__/screens/Swap/components/AnimatedSwapCoinIcon.tsx [hunk 21]
```diff
@@ -39,13 +40,13 @@ export const AmimatedSwapCoinIcon = React.memo(function FeedCoinIcon({
...
   const { isDarkMode, colors } = useTheme();
 
-  const imageLoadingError = useSharedValue(false);
+  const didErrorForUniqueId = useSharedValue<string | undefined>(undefined);
 
   const animatedIconSource = useAnimatedProps(() => {
     return {
       source: {
         ...DEFAULT_FASTER_IMAGE_CONFIG,
-        borderRadius: (small ? 16 : large ? 36 : 32) / 2,
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/AnimatedSwapCoinIcon.tsx:48]: This is the cause of https://linear.app/rainbow/issue/APP-1534/some-images-displayed-as-rectangles-on-android

File: src/__swaps__/screens/Swap/components/ReviewPanel.tsx [hunk 22]
```diff
@@ -111,9 +112,7 @@ export function ReviewPanel() {
 
   const unknown = i18n.t(i18n.l.swap.unknown);
 
-  const chainName = useDerivedValue(() =>
-    chainNameForChainIdWithMainnetSubstitutionWorklet(internalSelectedOutputAsset.value?.chainId ?? ChainId.mainnet)
-  );
+  const chainName = useDerivedValue(() => ChainNameDisplay[internalSelectedOutputAsset.value?.chainId ?? ChainId.mainnet]);
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/ReviewPanel.tsx:115]: should we pre-process chain names?

File: src/__swaps__/screens/Swap/components/SwapInputAsset.tsx [hunk 23]
```diff
@@ -15,17 +15,15 @@ import { TokenList } from '@/__swaps__/screens/Swap/components/TokenList/TokenLi
...
+import { AnimatedSwapCoinIcon } from './AnimatedSwapCoinIcon';
 
 function SwapInputActionButton() {
   const { isDarkMode } = useColorMode();
   const { SwapNavigation, internalSelectedInputAsset } = useSwapContext();
 
   const label = useDerivedValue(() => {
     const asset = internalSelectedInputAsset.value;
-    return asset?.symbol ?? '';
+    return asset?.symbol ?? (!asset ? 'Select' : '');
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/SwapInputAsset.tsx:26]: i18n

File: src/__swaps__/screens/Swap/components/SwapInputAsset.tsx [hunk 24]
```diff
@@ -74,27 +67,20 @@ function SwapInputIcon() {
...
-      isSameAssetWorklet(userAsset, {
-        address: asset.address,
-        chainId: asset.chainId,
-      })
-    );
-    return userAsset?.balance.display ?? 'No balance';
+    const hasBalance = Number(asset?.balance.amount) > 0 && asset?.balance.display;
+    const balance = (hasBalance && asset?.balance.display) || 'No Balance';
+
+    return asset ? balance : 'Token to Swap';
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/SwapInputAsset.tsx:83]: i18n

File: src/__swaps__/screens/Swap/components/SwapInputAsset.tsx [hunk 25]
```diff
@@ -74,27 +67,20 @@ function SwapInputIcon() {
...
-
-    const userAsset = userAssets.find(userAsset =>
-      isSameAssetWorklet(userAsset, {
-        address: asset.address,
-        chainId: asset.chainId,
-      })
-    );
-    return userAsset?.balance.display ?? 'No balance';
+    const hasBalance = Number(asset?.balance.amount) > 0 && asset?.balance.display;
+    const balance = (hasBalance && asset?.balance.display) || 'No Balance';
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/SwapInputAsset.tsx:81]: i18n

File: src/__swaps__/screens/Swap/components/SwapOutputAsset.tsx [hunk 26]
```diff
@@ -74,27 +68,23 @@ function SwapInputIcon() {
...
-    if (!asset) return 'No balance';
-
-    const userAsset = userAssets.find(userAsset =>
-      isSameAssetWorklet(userAsset, {
-        address: asset.address,
-        chainId: asset.chainId,
-      })
-    );
-    return userAsset?.balance.display ?? 'No balance';
+    return selectedAssetId ? selectedAssetBalance || 'No Balance' : 'Token to Buy';
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/SwapOutputAsset.tsx:87]: i18n

File: src/__swaps__/screens/Swap/components/SwapSlider.tsx [hunk 27]
```diff
@@ -353,12 +350,20 @@ export const SwapSlider = ({
...
       opacity,
     };
   });
 
+  const sellingOrBridgingLabel = useDerivedValue(() => {
+    const areBothAssetsSelected = internalSelectedInputAsset.value && internalSelectedOutputAsset.value;
+    const isBridging =
+      areBothAssetsSelected && internalSelectedInputAsset.value?.mainnetAddress === internalSelectedOutputAsset.value?.mainnetAddress;
+
+    return isBridging ? 'Bridging' : 'Selling';
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/SwapSlider.tsx:364]: i18n

File: src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts [hunk 28]
```diff
@@ -23,156 +22,161 @@ export interface AssetToBuySection {
...
-  const [toChainId, setToChainId] = useState(selectedOutputChainId.value);
-  const [assetToSellAddress, setAssetToSellAddress] = useState(
-    assetToSell.value?.[assetToSell.value?.chainId === ChainId.mainnet ? 'address' : 'mainnetAddress']
-  );
+  const [state, setState] = useState({
+    assetToSellAddress: assetToSell.value?.[assetToSell.value?.chainId === ChainId.mainnet ? 'address' : 'mainnetAddress'],
+    fromChainId: assetToSell.value?.chainId ?? undefined,
+    isCrosschainSearch: assetToSell.value ? assetToSell.value.chainId !== selectedOutputChainId.value : false,
+    toChainId: selectedOutputChainId.value ?? ChainId.mainnet,
+  });
```
Comment (walmat) [inline src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts:40]: what was the reason these were combined into a single state object? Generally a bigger fan of keeping stuff separated. See https://legacy.reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables

File: src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts [hunk 29]
```diff
@@ -23,156 +22,161 @@ export interface AssetToBuySection {
...
 
-  const query = useMemo(() => searchQuery.trim().toLowerCase(), [searchQuery]);
-  const enableUnverifiedSearch = useMemo(() => searchQuery.length > 2, [searchQuery]);
+  const debouncedStateSet = useDebouncedCallback(setState, 500, { leading: true, trailing: true });
```
Comment (walmat) [inline src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts:42]: see comment above, but this is going to be problematic since we're not keeping the previous reference to the old state.

File: src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts [hunk 30]
```diff
@@ -23,156 +22,161 @@ export interface AssetToBuySection {
...
-      if (previous !== current) {
-        runOnJS(setToChainId)(current);
-      }
-    }
+  const sliceTopResultsByNetworkWithCoinIconUrls = useCallback(
+    (data: TokenSearchResult) => {
+      const matchingNetwork = data.filter(asset => asset.chainId === state.toChainId);
+      return matchingNetwork.slice(0, 20).filter(asset => asset.icon_url);
+    },
+    [state.toChainId]
```
Comment (walmat) [inline src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts:66]: This is going to remove the other networks section as far as I can tell, haven't tested.

File: src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts [hunk 31]
```diff
@@ -23,156 +22,161 @@ export interface AssetToBuySection {
...
+      return mainnetAddress ?? token.address;
+    }
+    return token.networks[chainId].address;
+  }, []);
+
+  const { favoritesMetadata: rawFavorites } = useFavorites();
+  const favorites = useMemo(() => Object.values(rawFavorites), [rawFavorites]);
+
+  const unfilteredFavorites = useMemo(() => {
+    return Object.values(favorites)
```
Comment (walmat) [inline src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts:114]: you're calling Object.values twice here. Also think memoizing favorites is unnecessary since it's only being used in unfilteredFavorites which is already memoized.

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts (hunk 3 above)
Comment (walmat) [reply src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:94]: Follow-up question before I can answer that one, but if we're prefilling an asset to buy, are we prefilling the asset to sell as the largest asset on the same network?

In not: We can do this no problem and just leave the output values as `0`
If so: We'll have to prefetch a quote to show the initial output values based on that quote.

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts (hunk 4 above)
Comment (walmat) [reply src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:112]: nativePrice gets set via the fetchAssetPrices interval / calls.
asset price.value can be set if we previously had the price from it being in the userAssets (aka those assets include price data)

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts (hunk 6 above)
Comment (walmat) [reply src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:252]: Yep this is exactly why I was always setting the isFetching value here.

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 32]
```diff
@@ -438,13 +506,12 @@ export function useSwapInputsController({
   };
 
   const quoteFetchingInterval = useAnimatedInterval({
-    intervalMs: 10_000,
+    intervalMs: 12_000,
```
Comment (walmat) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:509]: should extract this value out and name is like ESTIMATED_BLOCK_TIME or something

**Review by walmat — APPROVED** [review]
